### PR TITLE
build: update next v14

### PR DESCRIPTION
**Pull request by Asuka109**
#### 💻 变更类型 | Change Type

- [ ] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [x] 🔨 chore
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

Hope this will help to resolve https://github.com/lobehub/lobe-chat/pull/381#issuecomment-1783985366

And the 4a3311d just used to suppress hydration warning caused by some common browser extensions (dark reader, cubox, etc.) that will inject scripts and mutate the root node.

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -1,5 +1,5 @@
+import nextPWA from '@ducanh2912/next-pwa';
```
Comment (arvinxx) [inline next.config.mjs:1]: why use this package? Is it better than `next-pwa`?

File: next.config.mjs [hunk 2]
```diff
@@ -68,15 +77,10 @@ const nextConfig = {
       layers: true,
     };
 
-    return config;
-  },
+    config.resolve.alias['@ant-design/cssinjs'] = '@ant-design/cssinjs/lib';
```
Comment (arvinxx) [inline next.config.mjs:80]: It's an awsome idea! I have tried with `transpilePackages` but find it not working correctly. 😅

cc @Madccc

File: next.config.mjs (hunk 1 above)
Comment (Asuka109) [reply next.config.mjs:1]: https://github.com/shadowwalker/next-pwa has not been actively maintained for months and does not work fine with next@14. Therefore, we need to switch to the forked version at https://github.com/DuCanhGH/next-pwa.

Comment (arvinxx) [reply next.config.mjs:1]: Good catch

File: next.config.mjs (hunk 2 above)
Comment (canisminor1990) [reply next.config.mjs:80]: 666

**Pull request merged**